### ISSUE TITLE
RuntimeWarning: invalid value encountered in log targets_dw = np.log(gt_widths / ex_widths) 

### ISSUE THREAD

**Issue by vikiboy** [rbgirshick/py-faster-rcnn#268]
While training py-faster-rcnn on a custom dataset following the instructions at https://github.com/deboc/py-faster-rcnn/blob/master/help/Readme.md

I encountered some errors like `AttributeError: 'numpy.ndarray' object has no attribute 'toarray' in py-faster-rcnn`

which I managed to bypass by editing the `roidb.py` 
```
gt_overlaps = roidb[i]['gt_overlaps']
gt_overlaps = sp.sparse.csr_matrix(gt_overlaps).toarray()

```

However, during the training process, I received a warning twice 

`RuntimeWarning: invalid value encountered in log targets_dw = np.log(gt_widths / ex_widths) `

in the file bbox_transform.py

Are the results going to be affected by this ?

**Comment by GeorgiAngelov** [comment]
I have the same question...

**Comment by ashwin** [comment]
@vikiboy `numpy.log` gives this warning when its input is negative. Maybe you want to use PDB and check the values of `gt_widths` and `ex_widths` and figure out why one of them is negative.

**Comment by HyunJun-Jo** [comment]
@ashwin I found that the gt_widths is negarive as you said. But I cannot find which training data have this error. Is there any way to confirm that?

**Comment by ck196** [comment]
@HyunJun-Jo I checked negative images by editing lib/datasets/imdb.py
Check all boxes elements > 0. if not (boxes > 0).all()
```
def append_flipped_images(self):
        num_images = self.num_images
        widths = self._get_widths()
        for i in xrange(num_images):
            boxes = self.roidb[i]['boxes'].copy()
            if not (boxes > 0).all():
                print self.image_path_at(i)
```

**Comment by zhaohb-tsinghua** [comment]
I have resolved the problem,,
in the function “_load_pascal_annotation(,)” the right one should be:
x1 = float(bbox.find('xmin').text)                                                                 
y1 = float(bbox.find('ymin').text)                                      
x2 = float(bbox.find('xmax').text)                                                                 
y2 = float(bbox.find('ymax').text)

**Comment by Designbook1** [comment]
@zhaohb-tsinghua can you tell me why the author wirte like this:
            ```
            # Make pixel indexes 0-based
            x1 = float(bbox.find('xmin').text) - 1
            y1 = float(bbox.find('ymin').text) - 1
            x2 = float(bbox.find('xmax').text) - 1
            y2 = float(bbox.find('ymax').text) - 1

```

**Comment by Designbook1** [comment]
@zhaohb-tsinghua can you tell me why the author wirte like this:

```
# Make pixel indexes 0-based
x1 = float(bbox.find('xmin').text) - 1
y1 = float(bbox.find('ymin').text) - 1
x2 = float(bbox.find('xmax').text) - 1
y2 = float(bbox.find('ymax').text) - 1
```

**Comment by zhaohb-tsinghua** [comment]
感觉作者应该是把matlab的东西从新使用，matlab最小索引是1开始，其他有些程序是从0开始。   我觉的就是在调整这种索引起始位置的问题，正好差一个像素，但是我们使用自己的数据的时候很有可能最小值为0，导致0减1为负数。

在2017-05-05 11:31:00，赵海波zhaohb@mail.tsinghua.edu.cn写道：

@zhaohb-tsinghua can you tell me why the author wirte like this:

# Make pixel indexes 0-based
x1 = float(bbox.find('xmin').text) - 1
y1 = float(bbox.find('ymin').text) - 1
x2 = float(bbox.find('xmax').text) - 1
y2 = float(bbox.find('ymax').text) - 1

**Comment by zhaohb-tsinghua** [comment]
Sorry. I sent the wrong message to you.  Please don't mind.

在2017-05-05 15:14:17，赵海波zhaohb@mail.tsinghua.edu.cn写道：

感觉作者应该是把matlab的东西从新使用，matlab最小索引是1开始，其他有些程序是从0开始。   我觉的就是在调整这种索引起始位置的问题，正好差一个像素，但是我们使用自己的数据的时候很有可能最小值为0，导致0减1为负数。

在2017-05-05 11:31:00，赵海波zhaohb@mail.tsinghua.edu.cn写道：

@zhaohb-tsinghua can you tell me why the author wirte like this:

# Make pixel indexes 0-based
x1 = float(bbox.find('xmin').text) - 1
y1 = float(bbox.find('ymin').text) - 1
x2 = float(bbox.find('xmax').text) - 1
y2 = float(bbox.find('ymax').text) - 1

**Comment by Designbook1** [comment]
@zhaohb-tsinghua I know matlab is 1-base and the labelled data may be 0-based, so when I transfer txt into xml, I write:
`xmin=xmin+1;
ymin=ymin+1`

**Comment by deatherving** [comment]
The answer is to check the bndbox. The starting coordinate of the box should not be 0. I set the lowest box coordinate to 1 and everything is fine.

**Comment by Designbook1** [comment]
@deatherving it can be 0, but cannot be negative or big number.

**Comment by deatherving** [comment]
@Designbook1 You are right. As you said, the "minus 1" caused the problem. So I set it to 1 and after minus 1 it is 0.

**Comment by Designbook1** [comment]
@deatherving In fact, our purpose is to enhance the robustness of the code from bad data. such as:
`00434400.jpg 0 -2 23 32` , we also protect non-negative sign in addition to non-negative. 
1.`0-1=65535`
2.-2 change to 2

**Comment by JustAnotherUsername123** [comment]
I had the same error as @vikiboy and tried to fix it with his method:
replacing
`gt_overlaps = roidb[i]['gt_overlaps'].toarray()`
with
```
gt_overlaps = roidb[i]['gt_overlaps']
gt_overlaps = sp.sparse.csr_matrix(gt_overlaps).toarray()
```
but than now I get the following error:

> [...]raise ValueError("The truth value of an array with more than one "
> ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all().

Has anybody had the same problem or a solution for this problem?

**Comment by CelineZhou03** [comment]
@zhaohb-tsinghua Where can I find this function “_load_pascal_annotation(,)”,  can you give a path?
